### DISCUSSION TITLE
Thanks, and a couple of questions on MQTT and the Climate  Control in HA

### DISCUSSION THREAD

**Discussion by Nick-Sills** [revk/xESP32-Faikout#326] [General]
First, thanks so much @revk for putting this together, and for all your hard work. This is an awesome project! After I discovered that Daikin had limited the number of polls to their web API I decided enough is enough, and found this repo. I've just installed two faikin boards in a FTXM42R and FTXM20R today.  I'll post some pics of my cable making and installation in another thread. 

I've got a couple of questions, if that's OK.

1. The Climate control in Home Assistant only has 'Idle' and 'Off' states, when ideally it should just be 'Auto', 'Heat', Cool' or 'Off', with no 'Idle. When running, the entities permanently show 'Idle', and the icons don't change colour with the state as they should do. 

<img width="361" alt="image" src="https://github.com/revk/ESP-Faikin/assets/162114112/1d4589a7-487f-4025-8557-827d5b02aff5">

2. The AC units are in a remote location, so I'm using an MQTT bridge over a Site-2-Site VPN, with topic filtering to keep entities and control managed. I can't seem to find a way to change the faikin base MQTT topic to match my site base topic. Is it possible?

Many thanks for any help

**Comment by revk** [comment]
The HVAC_Action is only set for *Faikin auto* as we have no way to find it from the unit directly that I know of.

The beta has a new option to change the topic order, to make the base topic the "hostname" for the device. Does that help. It is `prefixhost`

**Reply by Nick-Sills** [reply]
Thanks 

The native Daikin integration doesn't show the HVAC_Action for auto mode either, it just shows the 'mode' of 'Heat/Cool'.

Daikin integration:
<img width="350" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/162114112/99a0388a-ced4-42d1-a8c9-7ec38ad625e4">

If it were possible to not show 'idle' state for Faikin, then it would perhaps  be more visually appealing and colour the icons? 

I think I've managed to sort the MQTT prefix in the advanced settings by setting my own prefix:

![image](https://github.com/revk/ESP32-Faikin/assets/162114112/1eb0755d-9654-4306-9d3a-aaf815c7ecff)

I'm getting the states in Home Assistant, but commands don't seem to be working now.  

Thanks again

**Reply by revk** [reply]
Ah, yes, latest beta now allows `/` in the `command` and `setting` as well. That did not used to work.

**Reply by Nick-Sills** [reply]
> Ah, yes, latest beta now allows / in the command and setting as well. That did not used to work.

Great! I'll pull the beta down. 

Last question ( I hope :-) ).  

I've successfully added a Bluetooth temp sensor, but the 'Enable' button doesn't do anything, it just maintains 'Off'. 

![image](https://github.com/revk/ESP32-Faikin/assets/162114112/3edaeaf0-5deb-4337-b1a3-890cf4a028ed)

I've still got the Daikin WiFi board connected, and the Onecta app is still working. Is that why? I was wondering how the Faikin auto mode could possibly work with the Daikin auto mode active as well.

Thanks

**Reply by revk** [reply]
What, you can't set ½, 1, etc?

**Reply by Nick-Sills** [reply]
Sorry, I was being a dumb arse... I was clicking it and expecting it to change state,

**Reply by revk** [reply]
Phew

**Reply by Nick-Sills** [reply]
Hi @revk 
The  hvac_action string of "off" or "idle" makes it difficult to interpret the climate entity when just showing entity states rather than the climate control UI. It's currently showing state "Off (Cool)" when it's cooling. Would it be possible to send a fake or empty "hvac_action" to prevent "off" or "idle" being permanently shown in the entities card? 

Set hvac_action to "cooling" when the mode is "cool"
Set hvac_action to "heating" when the mode is "heat"
Set hvac_action to "off" when the mode is "off"
For other states an empty string.

Alternatively, just set hvac_action to an empty string always. I could then use the the card-mod integration to change icon colour based on mode. I've been playing with this and it works well. 

I'm still having a problem with setting the MQTT prefix with '/' for commands with the beta build. It's probably something I've done, so I'm investigating. 

Thanks for all your help!

**Comment by Elwell** [comment]
hah, I was just getting minorly annoyed at the 'idle' too - according to https://developers.home-assistant.io/docs/core/entity/climate the default is 'None'. Having this empty gets a +1 from me too, unless you're actively setting the tracking in faikin auto (which I don't use) and then you can set heating/cooling as applicable
